File: frontend/src/types/splits.ts
export interface SplitEntry {
  user_id: string;
  split_seconds: number;
}

export interface BeaconPairSplits {
  from_beacon_id: number | null;
  to_beacon_id: number;
  splits: SplitEntry[];
}

export interface CompetitorSummary {
  user_id: string;
  first_name: string;
  last_name: string;
  sex: string | null;
  course_number: number | null;
}

export interface EventBeaconInfo {
  id: number;
  number: number;
  tag: string;
  is_ph: boolean;
  code: string;
}

export interface SplitsResponse {
  event_name: string;
  beacons: EventBeaconInfo[];
  competitors: CompetitorSummary[];
  pairs: BeaconPairSplits[];
}

File: backend/src/app/schemas/splits.py
from pydantic import BaseModel

from app.schemas.events import EventBeacon


class CompetitorSummary(BaseModel):
    """Lightweight competitor info for split times display."""

    user_id: str
    first_name: str
    last_name: str
    sex: str | None = None
    course_number: int | None = None


class SplitEntry(BaseModel):
    """One competitor's split time for a beacon pair."""

    user_id: str
    split_seconds: int


class BeaconPairSplits(BaseModel):
    """Split times for one pair of consecutive beacons."""

    from_beacon_id: int | None = None  # None = Départ
    to_beacon_id: int
    splits: list[SplitEntry] = []


class SplitsResponse(BaseModel):
    """Full response for the split times endpoint."""

    event_name: str
    beacons: list[EventBeacon] = []
    competitors: list[CompetitorSummary] = []
    pairs: list[BeaconPairSplits] = []

